src/value.zig

```diff
@@ -956,6 +975,26 @@ pub const Value = struct {
                     .name = name,
                 } })).toValue();
             },
+            .Union => switch (ty.containerLayout(mod)) {
+                .Auto => return error.IllDefinedMemoryLayout,
+                .Extern => {
+                    const union_obj = mod.typeToUnion(ty).?;
+                    const largest_field = mod.unionLargestField(union_obj);
+                    const field_size: usize = @intCast(largest_field.size);
+                    const val = try (try readFromMemory(largest_field.ty, mod, buffer[0..field_size], arena)).intern(largest_field.ty, mod);
+                    return (try mod.intern(.{
+                        .un = .{
+                            .ty = ty.toIntern(),
+                            .tag = .undef,
```

Comment:
Why use `undef` for this? You're specifically parsing the largest field - the tag can be set appropriately.